encrypted-media/scripts/playback-temporary-waitingforkey.js

```diff
@@ -72,19 +54,18 @@ function runTest(config) {
             return access.createMediaKeys();
         }).then(function(mediaKeys) {
             _mediaKeys = mediaKeys;
-
             _video.setMediaKeys(_mediaKeys);
-
             waitForEventAndRunStep('waitingforkey', _video, onWaitingForKey, test);
-
+            
+            // Not using waitForEventAndRunStep() to avoid too many
```

Comment:
The previous location ensured that the test would never complete without seeing waitingforkey at least one time. The flow is cleaner here, but it could cover up a test or UA bug.